classes/file_manager.php

```diff
@@ -0,0 +1,186 @@
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace quizaccess_seb;
+
+use stored_file;
+
+defined('MOODLE_INTERNAL') || die();
+
+class file_manager {
+
+    /** Component the file will be stored in.  */
+    const COMPONENT = 'mod_quiz';
+
+    /** File area in the module to store files.  */
+    const FILE_AREA = 'quizaccess_seb_quizsettings';
+
+    /**
+     * Get a stored file object representing a draft file saved from a form.
+     *
+     * @param string $itemid
+     * @return stored_file|null
+     * @throws \coding_exception
+     */
+    public function get_form_file_by_itemid(string $itemid) {
+        global $USER; // Uploaded files initially have a user context.
+        $file  = null;
+
+        $fs = get_file_storage();
+        $context = \context_user::instance($USER->id);
+        $areafiles = $fs->get_area_files($context->id, 'user', 'draft', $itemid, 'id DESC', false);
+        if (count($areafiles) === 1) {
+            $file = reset($areafiles);
+        }
+        return $file;
+    }
+
+    /**
+     * Get a saved file from the safe exam browser file area, for a specific quiz.
+     *
+     * @param string $itemid
+     * @param string $cmid
+     * @return stored_file|null
+     * @throws \coding_exception
+     */
+    public function get_module_file_by_itemid(string $itemid, string $cmid) {
+        $file  = null;
+        $fs = get_file_storage();
+
+        if (get_coursemodule_from_id('quiz', $cmid) && !empty($itemid)) { // Validate course module id.
+            $context = \context_module::instance($cmid);
+            $areafiles = $fs->get_area_files($context->id, self::COMPONENT, self::FILE_AREA, $itemid, 'id DESC', false);
+            if (count($areafiles) === 1) {
+                $file = reset($areafiles);
+            }
+        }
+        return $file;
+    }
+
+    /**
+     * Delete a saved file from the safe exam browser file area, for a specific quiz.
+     *
+     * @param string $itemid
+     * @param string $cmid
+     * @return bool
+     * @throws \coding_exception
+     */
+    public function delete_module_file_by_itemid(string $itemid, string $cmid) : bool {
+        $file = $this->get_module_file_by_itemid($itemid, $cmid);
+        if (!empty($file)) {
+            return $file->delete();
+        }
+        return false;
+    }
+
+    /**
+     * Check if a file is a valid seb config file. The file may be unencrypted xml or encryped binary. If encrypted,
+     * the encryption password is required to decrypt it.
+     *
+     * @param stored_file $uploadedconfig
+     * @param string $password
+     * @return bool
+     */
+    public function is_file_valid_seb_config(stored_file $uploadedconfig, string $password = '') : bool {
+        $content = $uploadedconfig->get_content();
+        // Attempt to decrypt file contents.
+        $content = seb_cipher::decrypt($content, $password);
```

Comment:
do we decrypt regardless?

Reply:
Decrypt handles the case that the file doesn't need to be decrypted. It will just return the content straight away.